### PR TITLE
Cu 86c0c42k4 filters are not working

### PR DESCRIPTION
**Pull request by brett-onions**
## Summary by CodeRabbit

- **New Features**
	- Enhanced transaction log fetching with improved filtering options.
	- Filters now trigger log fetching automatically based on user selections.
	- Updated menu item text for clarity.

- **Bug Fixes**
	- Consolidated date filtering logic for more accurate results.

- **Documentation**
	- Updated method signatures in relevant components and interfaces to reflect new parameters.

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86c0c42k4 Filters are not working](https://app.clickup.com/t/86c0c42k4)

**Review by ItsMurumba — APPROVED** [review]
LGTM

**Comment by drizzentic** [comment]
@coderabbitai review

File: packages/transaction-log/src/components/common/app.main.component.tsx [hunk 1]
```diff
@@ -136,14 +136,24 @@ const App: React.FC = () => {
         )
 
         setTransactions(prevTransactions => {
-          const newTransactionListState = [...prevTransactions]
+          if(filteredResults){
+            return newTransactionsWithChannelDetails
+          }
```
[the remark this answers is not in the document]
Comment (brett-onions) [reply packages/transaction-log/src/components/common/app.main.component.tsx:141]: The state is over written so there should not be any duplicates coming from the server

**Review by marvinIsSacul — APPROVED** [review]
Looks alright.

**Pull request merged**